client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -57,25 +58,21 @@ const Checkout = React.createClass( {
 			return;
 		}
 
-		if ( this.props.cart.hasLoadedFromServer ) {
-			this.trackPageView();
-
-			if ( this.props.product ) {
-				this.addProductToCart();
-			}
+		if ( this.props.cart.hasLoadedFromServer && this.props.product ) {
+			this.addProductToCart();
 		}
 
 		window.scrollTo( 0, 0 );
 	},
 
 	componentWillReceiveProps: function( nextProps ) {
-		if ( ! this.props.cart.hasLoadedFromServer && nextProps.cart.hasLoadedFromServer ) {
-			// if the cart hadn't loaded when this mounted, record the page view when it loads
-			this.trackPageView( nextProps );
+		if ( ! this.props.cart.hasLoadedFromServer && nextProps.cart.hasLoadedFromServer && this.props.product ) {
+			this.addProductToCart();
+		}
 
-			if ( this.props.product ) {
-				this.addProductToCart();
-			}
+		if ( this.props.cart.hasPendingServerUpdates && ! nextProps.cart.hasPendingServerUpdates ) {
+			// We only want to track the page view when the cart has finished loading the products
+			this.trackPageView( nextProps );
```

Comment:
Yes, I can confirm that. It sends 3 events each time you delete an item from the cart:

![screen shot 2016-08-11 at 12 20 30](https://cloud.githubusercontent.com/assets/699132/17585779/065a602a-5fbe-11e6-8f6a-bbab129a4fc2.png)